### PR TITLE
Reorganize demos

### PR DESCRIPTION
**Pull request by ly015**
* Reorganize demo folder
* Fix out-of-date mmdet/mmtrack configs
* Fix a bug in inference_top_sown_pose_model which causes track_id missing (seems introduced by #560)

Remaining issues:
* Some video files used in example commands in demos do not exist

### CODE REVIEW

**Comment by innerlee** [comment]
I suggest putting scripts directly under demo/ folder. Other resources, including documentations can be organized into subfolders

**Comment by ly015** [comment]
Some videos for demos are not in the repo (e.g. demo_fly_video.avi, demo_macaque.mp4, demo_video.mp4). Are these just arbitrary file names to show the command usage, or we should provide actual video files?

**Comment by innerlee** [comment]
Yes they are not included. @jin-s13 we can provide their download urls maybe

**Comment by innerlee** [comment]
cv2 video capture supports url if I remembered correctly, we can check if the video filenames could be replaced by their links

File: demo/README.md [hunk 1]
```diff
@@ -7,43 +7,43 @@ This page provides tutorials about running demos. Please click the caption for m
 <div align="center">
     <img src="demo_coco.gif" width="600px" alt><br>
```
Comment (innerlee) [inline demo/README.md:8]: update img link

Comment (innerlee) [reply demo/README.md:8]: And img links in our repo's README

**Comment by innerlee** [comment]
the demo paths in https://github.com/open-mmlab/mmpose/blob/master/docs/merge_docs.sh#L11-L21 need to be updated as well

**Review by innerlee — APPROVED** [review]

**Pull request merged**